src/schema/sources.ts

```diff
@@ -347,56 +374,146 @@ const sourceToGQL = (source: Source): GQLSource => ({
 });
 
 export enum SourcePermissions {
-  View,
-  Post,
-  Leave,
-  Delete,
-  Edit,
+  View = 'view',
+  Post = 'post',
+  PostLimit = 'post_limit',
+  PostDelete = 'post_delete',
+  MemberRemove = 'member_remove',
+  ModeratorAdd = 'moderator_add',
+  ModeratorRemove = 'moderator_remove',
+  InviteDisable = 'invite_disable',
+  Leave = 'leave',
+  Delete = 'delete',
+  Edit = 'edit',
 }
 
+export const roleSourcePermissions: Record<
```

Comment:
Suggestion for improvement concat permissions of lower rank and add only the difference for higher ranks.
For example: Moderator is member + PostDelete, MemberRemove etc.
Might be easier to manage

Reply:
Great idea. I like this one.

Reply:
Please make sure it's not a function that runs on every access. We can do it only one time